Replicator/Pusher+Revs.cc

```diff
@@ -114,8 +129,11 @@ namespace litecore::repl {
 
             // Include the document history, but skip the current revision 'cause it's redundant
             alloc_slice history = request->historyString(doc);
-            if ( history.hasPrefix(fullRevID) && history.size > fullRevID.size )
+            if ( !fullReplacementRevID && history.hasPrefix(fullRevID) && history.size > fullRevID.size )
                 msg["history"_sl] = history.from(fullRevID.size + 1);
+            else if ( fullReplacementRevID && history.hasPrefix(fullReplacementRevID)
+                      && history.size > fullReplacementRevID.size )
+                msg["history"_sl] = history.from(fullReplacementRevID.size + 1);
```

Comment:
Code for fullRevID and fullReplacementRevID here are pretty much the same. It can be simplified a bit with something like :

```
alloc_slice effectiveRevID = fullReplacementRev ! = null ? fullReplacementRev : fullRevID;
// then the logic here will just use effectiveRevID
```